modules/chain-grabber/src/main/scala/org/ergoplatform/explorer/indexer/processes/ChainIndexer.scala

```diff
@@ -176,6 +179,30 @@ object ChainIndexer {
     private def markAsMain(id: Id, height: Int): F[Unit] =
       pendingChainUpdates.update(_ :+ (id -> height))
 
+    private def updateBlockInfo(prevBlockId: Id, blockId: Id): F[Unit] =
+      (for {
+        prevBlockStats    <- OptionT(getBlockInfo(prevBlockId))
+        currentBlockStats <- OptionT(getBlockInfo(blockId))
+        correctTotalBlockchainSize = prevBlockStats.blockChainTotalSize + currentBlockStats.blockSize
+        correctTotalTxsCount       = prevBlockStats.totalTxsCount + currentBlockStats.txsCount
+        blockMiningTime            = currentBlockStats.timestamp - prevBlockStats.timestamp
+        correctTotalMiningTime     = prevBlockStats.totalMiningTime + blockMiningTime
+        correctTotalFees           = prevBlockStats.totalFees + currentBlockStats.blockFee
+        correctMinerReward         = prevBlockStats.totalMinersReward + currentBlockStats.minerReward
+        correctTotalCoins          = prevBlockStats.totalCoinsInTxs + currentBlockStats.blockCoins
```

Comment:
Stats calculation logic shouldn't reside in Indexer. Moreover, it seems to be very similar to https://github.com/ergoplatform/explorer-backend/blob/375c21ac01fb7cc3a83ed15f0b08e5c338d14bd3/modules/chain-grabber/src/main/scala/org/ergoplatform/explorer/indexer/extractors/BlockInfoBuildFrom.scala#L55

So please avoid logic duplication